TACC/Core-CMS
# Task/fp 893 migrate cms guides & Task/fp 909 populate homepage

# Overview

Add hard-coded content into templates for Guides pages.

# Issues

- __[FP-893](https://jira.tacc.utexas.edu/browse/FP-893)__: Migrate Core v2 CMS Pages (i.e. GH-111)
    - [FP-750](https://jira.tacc.utexas.edu/browse/FP-750): Getting Started
    - [FP-751](https://jira.tacc.utexas.edu/browse/FP-751): Data Transfer Guides
    - ~~FP-961: Rewrite Data Transfer Guide~~ — maybe, to be done later
    - ~~FP-752: Portal User Guide~~ — maybe, to be done later
    - [FP-894](https://jira.tacc.utexas.edu/browse/FP-894): Portal Technology Stack
- [FP-909](https://jira.tacc.utexas.edu/browse/FP-909): Home Page (default content)

# Changes

- __New__: Add templates for hard-coded Guides pages and a homepage with default content.
- __New__: (Noop) Explicitly set `html` font size.¹
- __New__: Migrate font settings from CMS to Shared.
- __New__: Add CMS global spacing settings.
- __New__: Add new stylesheets: `s-guide-docs`, `s-inline-dl`, `u-nested-text-content`.
- __New__: Have CMS use Shared CSS settings.²
- __New__: Load styles for Guide template and Home template.

> ¹ This is preparation for future change to support `1rem` = `10px`. See [UI - Constants - Font](https://confluence.tacc.utexas.edu/x/cYAZCg) and [UI - Constants - Units](https://confluence.tacc.utexas.edu/x/dYAZCg).
> ² The `site_shared` should get its own repo; it houses assets shared by Portal and CMS (and Docs).

# Screenshots

<details><summary>FP-909 CMS Homepage Content</summary>

- Automatic\
![FP-909 CMS Home - Auto](https://user-images.githubusercontent.com/62723358/111200322-9c0f5480-858f-11eb-9a99-ff0a75917d5c.png)
- Custom\
![FP-909 CMS Home - Custom](https://user-images.githubusercontent.com/62723358/111200325-9ca7eb00-858f-11eb-9c83-63715fc6309e.png)

</details>
<details><summary>FP-893 Dropdown Navigation</summary>

![Getting Started](https://user-images.githubusercontent.com/62723358/112203883-030cb900-8be1-11eb-896b-545e5aac222e.png)
![Data Transfer Guide](https://user-images.githubusercontent.com/62723358/112039809-75629800-8b12-11eb-96fd-50a37a4acf8d.png)
![Globus Data Transfer Guide](https://user-images.githubusercontent.com/62723358/112039810-75fb2e80-8b12-11eb-9224-cf26d276489e.png)
![Portal Technology Stack](https://user-images.githubusercontent.com/62723358/112039813-7693c500-8b12-11eb-90b1-02d7350a0b0b.png)

</details>
<details><summary>FP-893 Page Renders</summary>

- [Getting Started](https://user-images.githubusercontent.com/62723358/112204001-2899c280-8be1-11eb-9b72-a656041c121f.jpeg)
- [Data Transfer Guide](https://user-images.githubusercontent.com/62723358/112038346-b9549d80-8b10-11eb-8206-53f34e3d458e.jpeg)
- [Globus Data Transfer Guide - Examples Hidden](https://user-images.githubusercontent.com/62723358/112038356-bbb6f780-8b10-11eb-911e-03b7f3212426.jpeg)
- [Globus Data Transfer Guide - Examples Shown](https://user-images.githubusercontent.com/62723358/112038362-bc4f8e00-8b10-11eb-9550-6b24e203a006.jpeg)
- [Portal Technology Stack](https://user-images.githubusercontent.com/62723358/112207724-79abb580-8be5-11eb-9157-e246f676c5d9.jpeg)

</details>

# Testing

## Re-Test After 45d94af

<details><summary>Update a CMS Set Up for FP-893 Before 45d94af</summary>

1. Update `secrets.py`: Replicate [change to `default_secrets.py`](https://github.com/TACC/Core-CMS/commit/45d94affc8184cde16936ae967e6d1cd1a676713#diff-cd12f1bf79989fb0b009477e13896318ca45300cd2044bda0e5981e6d4774de1).
2. Fix template load failure on Globus & Rsync:
    1. Visit `/admin/cms/page/`.
    2. Update templates for both "Globus Data Transfer Guide" & "Rsync Transfer Guide"
        1. Edit "Advanced Settings" (hold SHIFT and click ✎ button).
        2. Change the "TEMPLATE:" value to appropriate template.
        3. Save the change.
        4. Publish the page.\
        (On Admin page listing, click the blue circle; it should become green.)
3. (Optional) Update Title and Slug for "Rsync Transfer Guide":
    - "Rsync Data Transfer Guide" → "Data Transfer Guide"
    - `rsync-data-transfer-guide` → `data-transfer-guide`

</details>

## Fresh Testing

1. (Optional) For new CMS installs, follow [`README.md`][readme].
1. For existing CMS installs, follow [`README.md` Quick Start][readme-quick-start], where step 1 is performed by checking out the branch.
1. If using secrets other than the `default_secrets.py`,  add the new template files to the `_CMS_TEMPLATES` block.
    ```
    _CMS_TEMPLATES = (
        (...),
        ('home.html', 'Homepage'),
        ('guide.html', 'Guide'),
        ('guides/getting_started.html', 'Guide: Getting Started'),
        ('guides/globus_data_transfer.html', 'Guide: Globus Data Transfer'),
        ('guides/rsync_data_transfer.html', 'Guide: Rsync Data Transfer'),
        ('guides/portal_technology.html', 'Guide: Portal Technology Stack'),
    )
    ```
1. Add five pages, choose different new "Guide: …" Page Template for each, and test that each page³ __both__ (A) looks like—or better than—reference page⁴, __and__ (B) has content that is generic and accurate⁵.

    [Instructions (Screenshots)](https://github.com/TACC/Core-CMS/wiki/Add-CMS-Page-Using-Template)

    <details><summary>Page Reference Links</summary>

    - **Homepage**:
        - http://localhost:8000
        - [manually created (Core v2 Dev)](https://dev.cep.tacc.utexas.edu/)
        - [reference (Core v1)](https://cep-web-staging-01.tacc.utexas.edu/)
    - **Guide: Getting Started**:
        - http://localhost:8000/guides/getting-started/
        - [reference (UTRC)](https://utrc.tacc.utexas.edu/getting-started/)
    - **Guide: Globus Data Transfer**:
        - http://localhost:8000/guides/globus-data-transfer-guide/
        - [reference (Confluence)](https://confluence.tacc.utexas.edu/x/PAAlAw)
    - **Guide: Data Transfer**:
        - http://localhost:8000/guides/data-transfer-guide/
        - [reference (3dem)](https://3dem.org/help/rsync-data-transfer-guide/)
        - [reference (A2CPS)](https://a2cps-staging.tacc.utexas.edu/help/rsync-data-transfer-guide/)
    - **Guide: Portal Technology Stack**:
        - http://localhost:8000/guides/portal-technology-stack/
        - [reference (UTRC)](https://cep-web-staging-01.tacc.utexas.edu/help/portal-technology-stack/)

    </details>

[readme]: https://github.com/TACC/Core-CMS/blob/main/README.md
[readme-quick-start]: https://github.com/TACC/Core-CMS/blob/main/README.md#quick-start

# Notes

## Known Issues

- __all pages with images__: Images are using remote URLs. To be fixed via #147.
